conda-store-server/conda_store_server/action/generate_constructor_installer.py

```diff
@@ -53,11 +56,38 @@ def write_file(filename, s):
     # conda and pip need to be in dependencies for the post_install script
     dependencies = ["conda", "pip"]
     pip_dependencies = []
-    for d in specification.dependencies:
-        if type(d) is schema.CondaSpecificationPip:
-            pip_dependencies.extend(d.pip)
-        else:
-            dependencies.append(d)
+    channels = []
+
+    if is_lockfile:
+        # Adds channels
+        channels = [c.url for c in specification.lockfile.metadata.channels]
+
+        # Adds dependencies
+        for p in specification.lockfile.package:
+            # Ignores packages not matching the current platform. Versions can
+            # be different between platforms or a package might not support all
+            # platforms. constructor is cross-friendly, but we're currently
+            # building only for the current architecture, see the comment in
+            # get_installer_platform
+            if p.platform != get_installer_platform():
+                continue
+            if p.manager == "pip":
+                pip_dependencies.append(f"{p.name}=={p.version}")
+            else:
+                ext = ".tar.bz2" if p.url.endswith(".tar.bz2") else ".conda"
+                build_string = p.url[: -len(ext)].rsplit("-", maxsplit=1)[-1]
+                dependencies.append(f"{p.name}=={p.version}={build_string}")
+
+    else:
+        # Adds channels
+        channels = specification.channels
+
+        # Adds dependencies
+        for d in specification.dependencies:
+            if type(d) is schema.CondaSpecificationPip:
```

Comment:
```suggestion
            if isinstance(d, schema.CondaSpecificationPip):
```

Any reason to not use the more idiomatic `isinstance`?